sdk/src/wallet/operations/transaction/build_transaction.rs

```diff
@@ -48,6 +55,30 @@ where
             }
 
             inputs.push(Input::Utxo(UtxoInput::from(*input.output_id())));
+
+            if let Some(reward) = selected_transaction_data.mana_rewards.get(input.output_id()) {
+                mana_rewards += *reward;
+                context_inputs.insert(ContextInput::from(RewardContextInput::new(idx as _)?));
+                needs_commitment_context = true;
+            }
+        }
+
+        for output in selected_transaction_data
+            .outputs
+            .iter_mut()
+            .filter(|o| o.is_delegation())
+        {
+            // Created delegations have their start epoch set, and delayed delegations have their end set
+            if output.as_delegation().delegation_id().is_null() {
+                *output = DelegationOutputBuilder::from(output.as_delegation())
+                    .with_start_epoch(protocol_parameters.delegation_start_epoch(latest_slot_commitment_id))
+                    .finish_output()?;
+            } else {
+                *output = DelegationOutputBuilder::from(output.as_delegation())
+                    .with_end_epoch(protocol_parameters.delegation_end_epoch(latest_slot_commitment_id))
+                    .finish_output()?;
+            }
+            needs_commitment_context = true;
```

Comment:
creation AND transition need one, yeah ?

Reply:
I believe so